src/ameba/rule/style/redundant_parentheses.cr

```diff
@@ -0,0 +1,61 @@
+module Ameba::Rule::Style
+  # A rule that disallows redundant parentheses around control expressions.
+  #
+  # For example, this is considered invalid:
+  #
+  # ```
+  # if (foo = @foo)
+  #   do_something
+  # end
```

Comment:
could you please elaborate on the decision why this is considered invalid? According to the referenced ticket, parentheses should be required in this case. I believe the idea was to copy the rubocop's check in order to enforce a safe assignment:

https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Style/ParenthesesAroundCondition

<img width="391" alt="image" src="https://user-images.githubusercontent.com/3624712/201752554-444a11ae-0594-4259-8db8-275664bfd41d.png">

Reply:
Yeah, I was considering changing the example.

Reply:
This PR solves different case, the one you've mentioned is solved by #302.